processor/deltatocumulativeprocessor/internal/data/expo/merge.go

```diff
@@ -26,10 +26,10 @@ func Merge(arel, brel Buckets) {
 	size := up - lo
 
 	counts := pcommon.NewUInt64Slice()
-	counts.Append(make([]uint64, size-counts.Len())...)
+	counts.EnsureCapacity(size)
```

Comment:
How to reduce allocations by this optimization?

Reply:
It doesn't create the `[]uint64` that might cause allocation.